plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/QueryInsightsService.java

```diff
@@ -0,0 +1,216 @@
+        if (!enableCollect) {
+            throw new IllegalArgumentException("Cannot get query data when query insight feature is not enabled.");
+        }
+        clearOutdatedData();
+        List<R> queries = new ArrayList<>(store);
+        queries.sort(Collections.reverseOrder());
+        return queries;
+    }
+
+    /**
+     * Clear all outdated data in the store
+     */
+    public abstract void clearOutdatedData();
+
+    /**
+     * Reset the exporter with new config
+     *
+     * This function can be used to enable/disable an exporter, change the type of the exporter,
+     * or change the identifier of the exporter.
+     * @param enabled the enable flag to set on the exporter
+     * @param type The QueryInsightsExporterType to set on the exporter
+     * @param identifier the Identifier to set on the exporter
+     */
+    public abstract void resetExporter(boolean enabled, QueryInsightsExporterType type, String identifier);
+
+    /**
+     * Clear all data in the store
+     */
+    public void clearAllData() {
+        store.clear();
+    }
+
+    /**
+     * Set flag to enable or disable Query Insights data collection
+     * @param enableCollect Flag to enable or disable Query Insights data collection
+     */
+    public void setEnableCollect(boolean enableCollect) {
+        this.enableCollect = enableCollect;
+    }
+
+    /**
+     * Get if the Query Insights data collection is enabled
+     * @return if the Query Insights data collection is enabled
+     */
+    public boolean getEnableCollect() {
+        return this.enableCollect;
+    }
+
+    /**
+     * Set flag to enable or disable Query Insights data export
+     * @param enableExport
+     */
+    public void setEnableExport(boolean enableExport) {
+        this.enableExport = enableExport;
+    }
+
+    /**
+     * Get if the Query Insights data export is enabled
+     * @return if the Query Insights data export is enabled
+     */
+    public boolean getEnableExport() {
+        return this.enableExport;
+    }
+
+    /**
+     * Start the Query Insight Service.
+     */
+    @Override
+    protected void doStart() {
+        if (exporter != null && getEnableExport()) {
+            scheduledFuture = threadPool.scheduleWithFixedDelay(this::doExport, exportInterval, ThreadPool.Names.GENERIC);
+        }
+    }
+
+    /**
+     * Stop the Query Insight Service
+     */
+    @Override
+    protected void doStop() {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel();
+            if (exporter != null && getEnableExport()) {
+                doExport();
+            }
+        }
+    }
+
+    private void doExport() {
+        List<R> storedData = getQueryData();
+        try {
+            exporter.export(storedData);
+        } catch (Exception e) {
+            throw new RuntimeException(String.format(Locale.ROOT, "failed to export query insight data to sink, error: %s", e));
```

Comment:
what happens when export is failing and users are requesting for topN? 
will it show the stale data in topN? Will there be any warning in the response regarding staleness?

Reply:
Hi @rishabhmaurya , in order to keep the scope of this PR simple and not to do a lot of things in one PR, I'm removing the exporter logic.